pkg/doc/jose/jwk.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jose
+
+import "github.com/square/go-jose/v3"
+
+// JWK (JSON Web Key) is a JSON data structure that represents a cryptographic key.
+type JWK jose.JSONWebKey
+
+// JWK gets JWK from JOSE headers.
+func (h Headers) JWK() (*JWK, bool) {
+	jwkRaw, ok := h[HeaderJSONWebKey]
+	if !ok {
+		return nil, false
+	}
+
+	var jwk JWK
+
+	err := convertMapToValue(jwkRaw, &jwk)
+	if err != nil {
+		return nil, false
+	}
+
+	return &jwk, true
+}
+
+// ToDo/Code Review question: can this somehow be merged with the jose.JSONWebKey type?
```

Comment:
@Baha-sk

Reply:
if this represents the old JWK used in legacy and current JWE implementation, then no need to move it here... we'll use the Jose.JSONWebKey for the future crypto implementation.

Reply:
done